nodes/k8s_kind/k8s_kind.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2020 Nokia
+// Licensed under the BSD 3-Clause License.
+// SPDX-License-Identifier: BSD-3-Clause
+
+package k8s_kind
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/srl-labs/containerlab/clab/exec"
+	"github.com/srl-labs/containerlab/nodes"
+	"github.com/srl-labs/containerlab/runtime"
+	"github.com/srl-labs/containerlab/runtime/docker"
+	"github.com/srl-labs/containerlab/types"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"sigs.k8s.io/kind/pkg/apis/config/v1alpha4"
+	"sigs.k8s.io/kind/pkg/cluster"
+)
+
+var kindnames = []string{"k8s-kind"}
+
+// Register registers the node in the global Node map.
+func Register(r *nodes.NodeRegistry) {
+	r.Register(kindnames, func() nodes.Node {
+		return new(k8s_kind)
+	}, nil)
+}
+
+type k8s_kind struct {
+	nodes.DefaultNode
```

Comment:
you can add a field `labels map[string]string` to this object, and use it to store "containerlab" labels

Reply:
hmm, I think I've read this wrong, scratch this comment

Reply:
containerlab is a one-off tool. so there is no state kept between executions. and it just does what it is told ... e.g. deploy deploys the topology and quits ... exec the same... thats why we need to keep the state we need externally and we choose to do so in the labels of the nodes spawned by containerlab

Reply:
Let me see if I understand k8s_kind represent a cluster, i.e. multiple nodes, no?

Reply:
yes it does and that is a new thing. until k8s_kind there is a 1to1 mapping between configured node and resulting container

Reply:
I have the idea to spawn pause containers as placeholders for external and k8s_kind containers, to attach the labels to and do the necessary stuff or maintain an index of deployed nodes per topology in a seperate file or so all things that would also work but require a quite some change in clab rather then allowing user-defined labels on the kind docker containers.